modules/AlphaWalletFoundation/AlphaWalletFoundation/Transfer/eip1559/Eip1559FeeOracle.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Eip1559FeeOracle.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 16.08.2022.
+//
+
+import Foundation
+import BigInt
+import Combine
+
+actor Eip1559FeeOracle {
+    private let suggestEip1559: SuggestEip1559
+    private let blockchainProvider: BlockchainProvider
+
+    init(blockchainProvider: BlockchainProvider) {
+        self.blockchainProvider = blockchainProvider
+        self.suggestEip1559 = SuggestEip1559(blockchainProvider: blockchainProvider)
+    }
+
+    func eip1559FeeEstimates(blockCount: Int = 100, block: BlockParameter = .latest, rewardPercentile: [Int] = []) async throws -> GasEstimates {
+        do {
+            let feeHistory = try await blockchainProvider.feeHistory(blockCount: blockCount, block: block, rewardPercentile: rewardPercentile).first
```

Comment:
Seems like these 2 lines both call `BlockchainProvider.feeHistory(blockCount:block:rewardPercentile:)`, once directly and once indirectly. Seems like we only need 1 call?